includes/class-wc-payment-checker-maksuturva.php

```diff
@@ -154,10 +179,36 @@ public function check_payments( array $payments ) {
 	protected function log( $payment, array $response ) {
 		global $wpdb;
 
-		$wpdb->insert( $wpdb->prefix . self::TABLE_NAME, array(
-			'payment_id' => $payment->get_payment_id(),
-			'response'   => wp_json_encode( $response ),
-			'date_added' => date( 'Y-m-d H:i:s' ),
-		) ); // Db call ok.
+		$tbl        = $wpdb->prefix . self::TABLE_NAME;
+		$payment_id = $payment->get_payment_id();
+
+		// First get the results for the payment.
+		$results = $wpdb->get_results( $wpdb->prepare(
+			'SELECT payment_id, date_added, query_count FROM %s WHERE payment_id = %s', $tbl, $payment_id ) );
+
+		// By default we set query count to 1. Loop through any found results and increment the query_count.
+		$query_count = 1;
+		foreach ( $results as $result ) {
+			$query_count += $result->query_count;
+		}
+
+		// If we found anything, e.g. query_count is over 1, update the record and increase the query_count by one.
+		if ( $query_count > 1 ) {
+			$wpdb->update( $tbl, array(
+				'response'    => wp_json_encode( $response ),
+				'query_count' => $query_count,
+			), array(
+					'payment_id' => $payment_id,
```

Comment:
The payment_id is not indexed, will this not be slow:ish when the table size grows?